src/main/java/duke/taskmanagement/Storage.java

```diff
@@ -0,0 +1,207 @@
+        } catch (IOException e) {
+            System.out.println("Something went wrong: " + e.getMessage());
+        }
+    }
+
+    /**
+     * To load the text line by line from the file
+     * and store it into a List<Task>
+     * @return The List<Task> which each element is loaded from the text file.
+     */
+    public List<Task> loadData(){
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(filePath));
+            String line;
+            while ((line= reader.readLine()) != null) {
+                String[] parts = line.split(" \\| ");
+                String type = parts[0];
+                boolean isDone = parts[1].equals("1");
+                //System.out.println(isDone);
+                String description = parts[2];
+                Task task;
+
+                if (type.equals("T")) {
+                    task = new ToDo(description, isDone);
+                } else if (type.equals("E")) {
+                    String from = parts[3];
+                    String till = parts[4];
+                    task = new Event(description, from, till, isDone);
+                } else if (type.equals("D")) {
+                    String deadline = parts[3];
+                    task = new Deadline(description, deadline, isDone);
+                } else {
+                    continue;
+                }
+                ls.add(task);
+            }
+            reader.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return ls;
+    }
+
+    /**
+     * To delete a certain line in the text file
+     * @param lineIndexToDelete The index of the line (task) you
+     *                          want to delete.
+     */
+    public void deleteLine(int lineIndexToDelete) {
+        List<String> lines = new ArrayList<>();
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(filePath));
+            String line;
+            while((line=reader.readLine()) != null) {
+                lines.add(line);
+            }
+            reader.close();
+            if (lineIndexToDelete >= 0 && lineIndexToDelete-1 < lines.size()) {
+                lines.remove(lineIndexToDelete-1);
+                BufferedWriter writer = new BufferedWriter(new FileWriter(filePath));
+
+                for(String updatedLine : lines) {
+                    writer.write(updatedLine);
+                    writer.newLine();
+                }
+                writer.close();
+            } else {
+                System.out.println("invalid index");
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * To modify the text file as the user modify the
+     * status of the task to 'done'.
+     * @param index The index of task you wish to
+     *              change the status to done.
+     */
+    public void changeToDone(int index) {
+                List<String> lines = new ArrayList<>();
```

Comment:
Any reason for the wide indents you used?

Reply:
> Any reason for the wide indents you used?

Should this line require wide indents?